### PR TITLE
Support full-width looping videos in articles

### PR DESCRIPTION
**Pull request by simonbyford**
## What does this change?

Does two things:

1) sets the aspect ratio of the container for looping videos in articles by passing in `containerAspectRatio`. This follows on from https://github.com/guardian/dotcom-rendering/pull/15029
2) adds support for full-width self-hosted videos. Looping videos in articles should display at full-width. Looping videos on fronts should continue to display in a fixed-size box with letterboxing (i.e. not full-width)

## Why?

Editorial have requested it. There is a precedent because this is how images currently behave in articles.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| Screenshot 2025-12-23 at 12 06 51 | Screenshot 2025-12-23 at 12 00 12 |

### CODE REVIEW

File: dotcom-rendering/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -1013,6 +1013,7 @@ export const Card = ({
 									}
 									subtitleSize={subtitleSize}
 									enableHls={enableHls}
+									letterboxed={true}
```
Comment (RikRootsGuardian) [inline dotcom-rendering/src/components/Card/Card.tsx:1016]: I don't know why we're touching the Fronts card here? Is calculating the aspect ratio from the video itself not enough?

Another thought - what happened with the rumours that people were looking to add an aspect ratio attribute to media atoms?

Comment (simonbyford) [reply dotcom-rendering/src/components/Card/Card.tsx:1016]: Fair question, I've bundled together 2 distinct changes in this PR which is causing confusion. It probably should have been 2 PRs.

The first change uses the aspect ratio of the video to calculate the size of the video container for the purposes of reducing CLS. It has no impact on whether or not the video displays letterboxes. It just follows on from this PR where it was missed: https://github.com/guardian/dotcom-rendering/pull/15029

The second change introduces the concept of letterboxes which can be enabled or disabled. Since the default state is to disable letterboxes (see [Mat's comment](https://github.com/guardian/dotcom-rendering/pull/15056#issuecomment-3689293118)), we need to change the Fronts card to enable them again.

Comment (simonbyford) [reply dotcom-rendering/src/components/Card/Card.tsx:1016]: > Another thought - what happened with the rumours that people were looking to add an aspect ratio attribute to media atoms?

If I understand you correctly, [such a field exists now](https://github.com/guardian/content-atom/pull/184/changes#diff-6aeacd5f47329816862f7230ed4de6ae90871cca2fecc7100460151f2e190434R66). But it's a string (e.g. "5:4") and we need a numerical value so I opted to use the dimensions instead.

**Review by RikRootsGuardian — APPROVED** [review]
Happy to approve, following discussion.

File: dotcom-rendering/src/components/SelfHostedVideo.importable.tsx [hunk 2]
```diff
@@ -150,6 +153,7 @@ type Props = {
 	subtitleSource?: string;
 	subtitleSize: SubtitleSize;
 	enableHls: boolean;
+	letterboxed?: boolean;
```
Comment (domlander) [inline dotcom-rendering/src/components/SelfHostedVideo.importable.tsx:156]: What does `letterboxed` mean? Is it where the video can appear with grey bars at its left and right?

File: dotcom-rendering/src/components/SelfHostedVideo.importable.tsx [hunk 3]
```diff
@@ -55,20 +55,23 @@ const videoContainerStyles = (
...
-		 * The value "80" is derived from the aspect ratio of the 5:4 slot.
-		 * When other slots are used for self-hosted videos, this will need to be adjusted.
-		 */
-		max-width: ${aspectRatio * 80}%;
-	}
+	${letterboxed &&
+	css`
+		max-height: 100vh;
+		max-height: 100svh;
+		max-width: 100%;
```
Comment (domlander) [inline dotcom-rendering/src/components/SelfHostedVideo.importable.tsx:66]: Why are these three lines of CSS not useful for video on articles?

File: dotcom-rendering/src/components/SelfHostedVideo.importable.tsx (hunk 2 above)
Comment (simonbyford) [reply dotcom-rendering/src/components/SelfHostedVideo.importable.tsx:156]: That's right. I appreciate it isn't a standard or obvious term though and I'm open to changing it if you have any ideas? I originally went with `fullWidth` (to mean the opposite of `letterboxed`) but after some [feedback](https://github.com/guardian/dotcom-rendering/pull/15056#issuecomment-3689293118) I changed it.

File: dotcom-rendering/src/components/SelfHostedVideo.importable.tsx (hunk 3 above)
Comment (simonbyford) [reply dotcom-rendering/src/components/SelfHostedVideo.importable.tsx:66]: The requirement was that looping videos should take up the full width of an article, regardless of their aspect ratio. If you introduce a `max-height` then some very tall videos cannot take up the full width. For example:

Image

Source: https://viewer.code.dev-gutools.co.uk/preview/music/2025/dec/17/simon-looping-video-test

In reality this is an edge case and will probably never happen but that's what motivated me to make this change.

`max-width: 100%` should be fine though and can be moved out of the conditional.

File: dotcom-rendering/src/components/SelfHostedVideo.importable.tsx (hunk 2 above)
Comment (paperboyo) [reply dotcom-rendering/src/components/SelfHostedVideo.importable.tsx:156]: My fault for choosing the term itself (and, in our case, more correct would be, even more obscure, [pillarbox](https://en.wikipedia.org/wiki/Letterboxing_(filming)#Pillar-boxing_and_window-boxing)). But while I don’t care for the term much, I think the reasoning behind my earlier comment still stands? (whatever works, ofc, though!)

Comment (simonbyford) [reply dotcom-rendering/src/components/SelfHostedVideo.importable.tsx:156]: The reasoning is solid. The name is obscure but perhaps just a comment explaining it would clear things up?
